gtfs-realtime/proto/gtfs-realtime.proto

```diff
@@ -378,6 +420,18 @@ message TripUpdate {
     // be marked as schedule_relationship=SKIPPED.
     // NOTE: This field is still experimental, and subject to change. It may be formally adopted in the future. 
     optional string shape_id = 4;
+    
+    // Specifies the headsign for this trip when it differs from the original.
+    // NOTE: This field is still experimental, and subject to change. It may be formally adopted in the future.
+    optional string trip_headsign = 5;
+
+    // Specifies the name for this trip when it differs from the original.
+    // NOTE: This field is still experimental, and subject to change. It may be formally adopted in the future.
+    optional string trip_short_name = 6;
+
+    // Specifies the block for this trip when it differs from the original.
+    // NOTE: This field is still experimental, and subject to change. It may be formally adopted in the future.
+    optional string block_id = 7;
```

Comment:
Do we have a realtime data producer for this kind of data? If not I would consider this a speculative feature.

Reply:
I need trip headsigns and I intend to produce it as soon as possible.

Reply:
I have removed block id from this PR as I am not going to use it.